README.md

```diff
@@ -144,10 +144,12 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are some of the advantages/disadvantages of writing JavaScript code in a language that compiles to JavaScript?
 * What tools and techniques do you use debugging JavaScript code?
 * What language constructions do you use for iterating over object properties and array items?
+* Explain destructuring assignment.
 * Explain the difference between mutable and immutable objects.
   * What is an example of an immutable object in JavaScript?
   * What are the pros and cons of immutability?
   * How can you achieve immutability in your own code?
+* What is a pure function? What are the side effects of a pure function?
```

Comment:
@strugee, yes that is true. But if we ask the question:

>   What is a closure, and how/why would you use one?

The flip side of that question would be:

>   How would you _not_ use a closure?

Which then leads to:

>   What is a pure function?

>   What is a side effect?

Good examples of why this is needed or how it relates to clousure:
*   https://davidwalsh.name/preventing-sideeffects-javascript
*   https://medium.com/javascript-scene/the-single-biggest-mistake-programmers-make-every-day-62366b432308